pkg/controller/controller.go

```diff
@@ -742,6 +758,19 @@ func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.Persist
 		deletionAnnSecrets.namespace = provisionerSecretRef.Namespace
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass) && vacName != nil {
+		vac, err := p.client.StorageV1alpha1().VolumeAttributesClasses().Get(ctx, *vacName, metav1.GetOptions{})
+		if err != nil {
+			return nil, controller.ProvisioningNoChange, err
```

Comment:
We were saying that even if the VAC does not exist yet, we still want to make sure later to retry provisioning. I just noticed that there seems no appropriate ProvisioningState right now for this. The closest is ProvisioningReschedule . Should we defer this to beta? @msau42

Reply:
Seems like the reconciling is happening in https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/controller/controller.go we may want to make some changes there too.

Reply:
Would `ProvisioningInBackground` be better here? By my reading that should always cause a re-drive of provisioning after some delay.

Reply:
Thinking about it more deeply, I agree. As far as I know, there is no way to prevent a retry here - the external-provisioner will _always_ eventually retry. And as this error is potentially temporal (for example, if a networking issue blocks our access to the k8s api briefly), there is no way to know if a provisioning action is already ongoing or not.